src/nl/hannahsten/texifyidea/settings/conventions/TexifyConventionsScheme.kt

```diff
@@ -0,0 +1,70 @@
+package nl.hannahsten.texifyidea.settings.conventions
+
+import nl.hannahsten.texifyidea.lang.DefaultEnvironment.*
+import nl.hannahsten.texifyidea.lang.commands.LatexGenericRegularCommand.*
+import nl.hannahsten.texifyidea.lang.commands.LatexListingCommand.LSTINPUTLISTING
+import nl.hannahsten.texifyidea.util.magic.env
+
+/**
+ * A scheme instance stores settings for a specific scope.
+ *
+ * Concrete scheme implementations decide on which scopes make sense. For example, a scheme could represent different
+ * color settings for code or different inspection rules. In the context of Texify conventions, a scheme stores
+ * either project or IDE level conventions.
+ *
+ * Instances must be serializable since they are persisted as part of [TexifyConventionsGlobalState] or
+ * [TexifyConventionsProjectState].
+ */
+data class TexifyConventionsScheme(
```

Comment:
It's great to have schemes, I think that will work much better than the separate project settings that we used to have.

It looks like the implementation is very targeted towards conventions settings. Why is that, isn't this more generally applicable to the other TeXiFy settings (which are currently global settings) as well?

Reply:
> It's great to have schemes, I think that will work much better than the separate project settings that we used to have.

I am glad you like it. 

> It looks like the implementation is very targeted towards conventions settings. Why is that, isn't this more generally applicable to the other TeXiFy settings (which are currently global settings) as well?

The only reason it is currently targeted towards conventions is that they are the first use case I could think of. The principle is definitely applicable to other TeXiFy settings as well. One would have to think carefully in which contexts a project/global scheme or more generally a scheme A/scheme B/scheme C separation makes sense. That is, which settings are global, which ones vary from project to project (e.g. IntelliJ's inspection schemes) and which ones should have completely independent schemes (e.g. IntelliJ's colour schemes).

Most of the conventions settings code can be generalized with generics, leaving the conventions as an instantiation of a more general infrastructure. If you think that is something that could be useful in the future I can extract general interfaces and classes from the existing code.

Reply:
No, don't generalise more than necessary. It's just good to have in mind where it could be applicable to use this, but for now let's don't make it more complicated than it already is.